docs/research/blake-256-hash-function.md

```diff
@@ -1,13 +1,302 @@
+|---------------|--------------|-------|--------|
+|     256-bit   |   512-bit    |128-bit|64 -bit |
+
+The structure then works in three steps:
+
+1) Initialization of a large internal state represented by 16 words in a 4x4 word matrix.
+
+2) Iteration of 14 rounds, each round that calls the core function (**G** as explained in the next section) 8 times to transform this state.
+
+3) Finalization after a series of rounds, a new sequence of values taken from the matrix using sequence of initial values and salt. 
+
+A more detailed description of the compression function can be found in [SHA-3 proposal BLAKE](https://decred.org/research/aumasson2010.pdf).
+
+#### **ChaCha inspired core function**
+
+The core function of BLAKE is a variation of the stream cipher ChaCha by Bernstein, one of the fastest stream ciphers. ChaCha inspired the creators of BLAKE due to its simplicity and security [^1]. The core function called the **G** function is repeated 112 times in BLAKE-256 over 14 rounds with each round consisting of eight round-dependent transformations **G<sub>0</sub>**, · · · , **G<sub>7</sub>**. The **G** function of BLAKE-256 for index **i**, where **σ<sub>r</sub>** is a fixed permutation used in
+round **r**, **M<sub>σ<sub>r</sub></sub>** are message blocks and **C<sub>σ<sub>r</sub></sub>**
+are round-dependent constants (see appendix for more information on permutations and constants). The **G<sub>i</sub>** updates 4, 32-bit state words (a, b, c, d) twice as follows:
+
+
+>a ← a + b + (m<sub>σ<sub>r</sub></sub>(2i) ⊕ c<sub>σ<sub>r</sub></sub>(2i+1))
+
+>d ← (d ⊕ a) >>> 16
+
+>c ← c + d
+
+>b ← (b ⊕ c) >>> 12
+
+>a ← a + b + (m<sub>σ<sub>r</sub></sub>(2i+1) ⊕ c<sub>σ<sub>r</sub></sub>(2i))
+
+>d ← (d ⊕ a) >>> 8
+
+>c ← c + d
+
+>b ← (b ⊕ c) >>> 7
+
+**<em>Figure 4: G Function, based on ChaCha [^1]</em>**
+
+|Symbol|Meaning|
+|------|-------|
+|←|Variable assignment|
+|+|Addition Modulo 2<sup>32</sup>|
+|⊕|Boolean exclusive OR (XOR)|
+|>>>|k K-bit rotation to the right|
+|<<<|k K-bit rotation to the left
+
+**<em>Table 2: Operations symbols used to describe the **G** function</em>**
+
+The **G** function is inspired from the “quarter-round” function of the stream cipher ChaCha, 
+with the difference being the addition of the message words, and the reversing of the rotation directions. ChaCha transforms (a, b, c, d) as follows:
+
+>a ← a + b
+
+>d ← (d ⊕ a) <<< 16
+
+>c ← c + d
+
+>b ← (b ⊕ c) <<< 12
+
+>a ← a + b
+
+>d ← (d ⊕ a) <<< 8
+
+>c ← c + d
+
+>b ← (b ⊕ c) <<< 7
+
+**<em>Figure 5: ChaCha Function [^1]</em>**
+
+### **Security Evaluation Criteria**
+
+The security of a cryptographic hash function is determined by the number of queries or guesses required to solve the following problems:
+
+1) Pre-image resistance (one-way): Given the **_y_** as an output of the hash function, it is computationally infeasible to find message **_x_** such that **_h(x) = y_**. [^4]
+
+![Preimage resistance](/img/Preimage resistance.svg)
```

Comment:
When you use `1.` for numbered lists, you will need to indent this line by four spaces to ensure it is included in the previous list item